streaming/src/main/scala/org/apache/gearpump/streaming/appmaster/AppMaster.scala

```diff
@@ -42,30 +42,36 @@ class AppMaster(appContext : AppMasterContext, app : Application)  extends Appli
   private val LOG: Logger = LogUtil.getLogger(getClass, app = appId)
   LOG.info(s"AppMaster[$appId] is launched by $username $app xxxxxxxxxxxxxxxxx")
 
-  private val (clockService, taskManager, executorManager) = {
+  private val (taskManager, executorManager) = {
     val dag = DAG(userConfig.getValue[Graph[TaskDescription, Partitioner]](AppDescription.DAG).get)
-    val clockService : ActorRef = {
-      val store = new InMemoryAppStoreOnMaster(appId, masterProxy)
-      context.actorOf(Props(new ClockService(dag, store)))
-    }
+
     val executorManager = context.actorOf(ExecutorManager.props(userConfig, appContext))
-    val taskManager = context.actorOf(Props(new TaskManager(appContext, dag, clockService, executorManager, self)))
-    (clockService, taskManager, executorManager)
+
+    val store = new InMemoryAppStoreOnMaster(appId, appContext.masterProxy)
+    val clockService = context.actorOf(Props(new ClockService(dag, store)))
+
+    val taskScheduler: TaskScheduler = new TaskSchedulerImpl(appId, context.system.settings.config)
+    val taskManager = context.actorOf(Props(new TaskManager(appContext.appId, dag,
+      taskScheduler, executorManager, clockService, self)))
+    (taskManager, executorManager)
   }
```

Comment:
Hi @huafengw 

Using the AppMaster to mediate the call will make the code more complex. Other app should develop its own AppMaster, instead of reusing the streaming appmaster. I added private [appmaster] for all classes under appmaster to reject any potential reference by unknown code.